src/generic-container/started-generic-container.ts

```diff
@@ -11,20 +11,42 @@ import { Readable } from "stream";
 import { containerLogs } from "../docker/functions/container/container-logs";
 import { StoppedGenericContainer } from "./stopped-generic-container";
 import { stopContainer } from "../docker/functions/container/stop-container";
+import { restartContainer } from "../docker/functions/container/restart-container";
+import { WaitStrategy } from "../wait-strategy";
 
 export class StartedGenericContainer implements StartedTestContainer {
   constructor(
     private readonly container: Dockerode.Container,
     private readonly host: Host,
     private readonly inspectResult: InspectResult,
     private readonly boundPorts: BoundPorts,
-    private readonly name: ContainerName
+    private readonly name: ContainerName,
+    private readonly waitStrategy: WaitStrategy
   ) {}
 
   public async stop(options: Partial<StopOptions> = {}): Promise<StoppedTestContainer> {
     return this.stopContainer(options);
   }
 
+  public async restart(options: Partial<RestartOptions> = {}): Promise<StartedGenericContainer> {
+    const resolvedOptions: RestartOptions = { timeout: 0, ...options };
+    await restartContainer(this.container, resolvedOptions);
+    const newInspectRes = await inspectContainer(this.container);
+    const newBoundPorts = await BoundPorts.fromInspectResult(newInspectRes).filter(
```

Comment:
Do the bound ports change when restarting the container?

Reply:
Yes, after restarting it, it seems that a new port is assigned to them.